src/main/java/uk/gov/companieshouse/officerssearch/subdelta/search/IdExtractor.java

```diff
@@ -8,17 +8,19 @@
 import org.springframework.util.StringUtils;
 import uk.gov.companieshouse.logging.Logger;
 import uk.gov.companieshouse.logging.LoggerFactory;
+import uk.gov.companieshouse.officerssearch.subdelta.exception.NonRetryableException;
+import uk.gov.companieshouse.officerssearch.subdelta.logging.DataMapHolder;
 
 @Component
-class IdExtractor {
+public class IdExtractor {
 
     private static final Logger LOGGER = LoggerFactory.getLogger(NAMESPACE);
     private static final Pattern OFFICER_ID_PATTERN =
             Pattern.compile("(?<=officers/)([^/]+)(?=/appointments)");
     private static final String NULL_EMPTY_URI =
             "Could not extract company number from empty or null resource uri";
     private static final String EXTRACTION_ERROR =
-            "Could not extract company number from resource URI: ";
+            "Could not extract company number from resource URI: %s";
```

Comment:
Error messages here should refer to extracting an officer ID not a company number

Reply:
done